### ISSUE TITLE
transactions_block_hash_fkey constraint violations

### ISSUE THREAD

**Issue by sabondano** [poanetwork/blockscout#796]
```
00:02:13.155 [error] Task #PID<0.13617.440> started from Indexer.Block.Realtime.Fetcher terminating
** (Postgrex.Error) ERROR 23503 (foreign_key_violation): update or delete on table "blocks" violates foreign key constraint "transactions_block_hash_fkey" on table "transactions"

    table: transactions
    constraint: transactions_block_hash_fkey

Key (hash)=(\x6430880db2ccf38a5ada6b46d7d6efbdd01718205545fd8d60a574406344b2cc) is still referenced from table "transactions".
    (explorer) lib/explorer/repo.ex:58: anonymous fn/5 in Explorer.Repo.safe_insert_all/3
    (elixir) lib/enum.ex:1925: Enum."-reduce/3-lists^foldl/2-0-"/3
    (explorer) lib/explorer/chain/import.ex:837: Explorer.Chain.Import.insert_changes_list/2
    (explorer) lib/explorer/chain/import.ex:646: Explorer.Chain.Import.insert_blocks/2
    (ecto) lib/ecto/multi.ex:421: Ecto.Multi.apply_operation/5
    (elixir) lib/enum.ex:1925: Enum."-reduce/3-lists^foldl/2-0-"/3
    (ecto) lib/ecto/multi.ex:411: anonymous fn/5 in Ecto.Multi.apply_operations/5
    (ecto) lib/ecto/adapters/sql.ex:576: anonymous fn/3 in Ecto.Adapters.SQL.do_transaction/3
Function: #Function<12.57269569/0 in Indexer.Block.Realtime.Fetcher.start_fetch_and_import/3>
    Args: []
```

**Comment by sabondano** [comment]
🤔after chatting with @alexgaribay we're thinking these are reorgs.

**Comment by sabondano** [comment]
@acravenho Happy to work on it 😃!

**Comment by sabondano** [comment]
Quick update: We've decided to wait on adding reorg support until after uncles support lands on master.

**Comment by pasqu4le** [comment]
This issue was solved by properly supporting reorgs in #833, that also contains regression tests for transactions.
